include/aws/event-stream/event_stream_rpc.h

```diff
@@ -0,0 +1,41 @@
+#ifndef AWS_EVENT_STREAM_RPC_H
+#define AWS_EVENT_STREAM_RPC_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/event-stream/event_stream.h>
+
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_message_type_name;
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_message_flags_name;
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_stream_id_name;
+extern AWS_EVENT_STREAM_API const struct aws_byte_cursor aws_event_stream_rpc_operation_name;
+
+enum aws_event_stream_rpc_message_type {
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_APPLICATION_MESSAGE,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_APPLICATION_ERROR,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PING,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PING_RESPONSE,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_CONNECT,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_CONNECT_ACK,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_PROTOCOL_ERROR,
+    AWS_EVENT_STREAM_RPC_MESSAGE_TYPE_SERVER_ERROR,
```

Comment:
Seems like PROTOCOL_ERROR and SERVER_ERROR could combined into one message type. The client is going to react the same way regardless.

that's not true. The way a client responds to a protocol error is to fix whatever is broken, but one protocol error is not retryable. server error is retryable because something not the client's fault occured.

Reply:
I meant more that: They are both final state-machine-ending messages that terminate the client-connection. So why not combine them into 1 message type and put the error-code inside? The particular error-code within (protocol vs server error) can affect decisions made at a layer outside the connection itself (whether or not to retry).